lib/src/signed_video_h26x_auth.c

```diff
@@ -964,3 +967,48 @@ signed_video_add_nalu_and_authenticate(signed_video_t *self,
 
   return svi_rc_to_signed_video_rc(status);
 }
+
+SignedVideoReturnCode
+signed_video_set_public_key(signed_video_t *self, const char *public_key, size_t public_key_size)
+{
+  if (!self || !public_key || public_key_size == 0) return SV_INVALID_PARAMETER;
+  if (self->signature_info->public_key) return SV_NOT_SUPPORTED;
+
+  // Return SV_NOT_SUPPORTED if not called from start of stream
+  if (self->authentication_started) return SV_NOT_SUPPORTED;
+
+  EVP_PKEY *pkey = NULL;
```

Comment:
When I think about it, shouldn't the OpenSSL stuff be moved to signed_video_openssl.c?
It is preferable to keep this file without dependency of OpenSSL.